spec/js/post_office_search_spec.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { getByLabelText, getByRole, waitFor } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+const wait = (ms: number = 0) =>
+  new Promise<void>((res) => {
+    setTimeout(() => {
+      res();
+    }, ms);
+  });
+
+describe('Post Office Search', () => {
+  // This is simulated since the test env doesn't support fetch
+  const testServerPort = 9835;
+  const addressSearchUrl = `http://127.0.0.1:${testServerPort}/api/address_search`;
+  const locationsSearchUrl = `http://127.0.0.1:${testServerPort}/api/usps_locations`;
+
+  beforeEach(() => {
+    if ('fetch' in global) {
+      jest.spyOn(global, 'fetch');
+    } else {
+      // @ts-ignore
+      global.fetch = jest.fn();
+    }
```

Comment:
When is `global.fetch` not present?

I'm expecting that since we're running Node v18, we should have access to `fetch` everywhere.

Reply:
It appears that Jest didn't add support for using `fetch` until version 28, and even then it may not be available when using `jsdom`. See https://github.com/jestjs/jest/issues/13834.